### DISCUSSION TITLE
Why no data shown in the report after upgrade to 4.13 from 4.4.2 (RedHat JDK 11) even without exeption?

### DISCUSSION THREAD

**Discussion by davidwwd2011** [eclipse-birt/birt#1515] [Q&A] [answered]
Previous running report  with Birt 4.4.2 is recently upgraded to 4.14.0 (also tried 4.13.0). Though View report in Eclipse shows the report with data correctly. But when running the applicaiton, empty report is displayed but no data is get into the report even though with same parameters. No exceptions reported  except some warnings:

NFO: register fonts in /usr/share/fonts/TTF cost:0ms
Dec 04, 2023 10:59:55 PM org.eclipse.birt.report.engine.layout.pdf.font.FontMappingManagerFactory registerFontPath
INFO: register fonts in /usr/openwin/lib/locale/zh_TW/X11/fonts/TrueType cost:0ms
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.lowagie.text.pdf.MappedRandomAccessFile$1 (file:/C:/Users/WangD/.m2/repository/com/lowagie/itext/2.1.7/itext-2.1.7.jar) to method java.nio.DirectByteBuffer.cleaner()
WARNING: Please consider reporting this to the maintainers of com.lowagie.text.pdf.MappedRandomAccessFile$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Dec 04, 2023 10:59:59 PM org.eclipse.birt.report.engine.layout.pdf.font.FontMappingManagerFactory registerFontPath

Have been strugling some time but not be able to find a solution.
Your kind help will be highly appreciated. Thnaks in advance.

**Comment by hvbtup** [comment]
Are you sure that you are using BIRT 4.14.0 and removed the BIRT 4.4.2 libs? Is your runtime environment clean?
For example, BIRT 4.14 does not use iText anymore. It has been replaced with OpenPDF.

**Reply by davidwwd2011** [reply, marked as answer]
Thannk you, @hvbtup, for the reply. I have replace all the itext 2.1.7 with the OpenPDF  (com.github.librepdf) 1.3.33. Now no warning anymore. However, the report is still empty when running applicaiton (though template View PDF works).  There are some font related INFO messages (using 4.13):

INFO: load font config in jar:file:/C:/Users/usrD/.m2/repository/birt_runtime/org/eclipse/birt/org.eclipse.birt.runtime/4.13.0/org.eclipse.birt.runtime-4.13.0.jar!/fontsConfig.xml cost 35ms
Dec 11, 2023 3:46:44 PM org.eclipse.birt.report.engine.layout.pdf.font.FontMappingManagerFactory$2 run
INFO: register fonts in /usr/openwin/lib/locale/ko.UTF-8/X11/fonts/TrueType cost:0ms
Dec 11, 2023 3:46:44 PM org.eclipse.birt.report.engine.layout.pdf.font.FontMappingManagerFactory$2 run
INFO: register fonts in /usr/openwin/lib/locale/zh/X11/fonts/TrueType cost:0ms
Dec 11, 2023 3:46:44 PM org.eclipse.birt.report.engine.layout.pdf.font.FontMappingManagerFactory$2 run
INFO: register fonts in /var/lib/defoma/x-ttcidfont-conf.d/dirs/TrueType cost:0ms
Dec 11, 2023 3:46:44 PM org.eclipse.birt.report.engine.layout.pdf.font.FontMappingManagerFactory$2 run
INFO: register fonts in /usr/openwin/lib/locale/KOI8-R/X11/fonts/TrueType cost:0ms
....

Since no official latest Eclipse Birt Maven runtime is available, I use @siepkes'  (https://gist.github.com/siepkes/00f0c410933271b4f4e0d5eab18d7469) for install the related  jars.

Do I miss anything? Kind advices would be highly appreciated.

**Reply by hvbtup** [reply]
I'm sorry I can't help you. I never experienced such an issue.
It is probably a result of your custom environment or your application just doesn't find anything.

Most existing reports should work with BIRT 4.14 without a need to change anything.
A known exception to this rule is Javascript code in your BIRT report that tries to use custom Java classes.
Due to the Java module system they may result in illegal reflective access operations. These result in exceptions with Java 17 (no longer just warnings).

**Comment by hvbtup** [comment]
And BIRT 4.14 requires Java 17...

**Reply by davidwwd2011** [reply]
I am now using 4.13.
